x/evm/handler.go

```diff
@@ -14,16 +14,27 @@ import (
 
 // NewHandler returns a handler for Ethermint type messages.
 func NewHandler(k Keeper) sdk.Handler {
-	return func(ctx sdk.Context, msg sdk.Msg) (*sdk.Result, error) {
+	return func(ctx sdk.Context, msg sdk.Msg) (result *sdk.Result, err error) {
+		snapshotStateDB := k.CommitStateDB.Copy()
+		defer func() {
+			if r := recover(); r != nil {
+				k.CommitStateDB = snapshotStateDB
+				panic(r)
+			}
+		}()
```

Comment:
this requires a large comment to explain what's happening

Reply:
ok